### PR TITLE
Log draw loop execution time

### PR DESCRIPTION
**Pull request by balderdash**
Measure the median draw loop execution time using `performance.mark` and `performance.measure`, along with the average number of sprites on the screen every draw loop, and log that data to firehose.

### CODE REVIEW

File: apps/src/gamelab/GameLab.js [hunk 1]
```diff
@@ -1415,6 +1433,44 @@ GameLab.prototype.completeSetupIfSetupComplete = function () {
...
+      console.error(e);
+    }
+  }
+};
+
+GameLab.prototype.measureDrawLoop = function (name, callback) {
+  mark(`${name}_start`);
+  callback();
+  measure(name, `${name}_start`);
+  this.spriteTotalCount += this.gameLabP5.p5.allSprites.length;
```
Comment (joshlory) [inline apps/src/gamelab/GameLab.js]: This re-counts the sprites every frame?

File: apps/src/gamelab/GameLab.js [hunk 2]
```diff
@@ -1423,36 +1479,10 @@ GameLab.prototype.onP5Draw = function () {
...
-            this.onPuzzleComplete(false, testResult);
-          } else if (validationResult === 'failed') {
-            // TODO(ram): Show failure feedback
-          }
-        } catch (e) {
-          // If validation code errors, assume it was neither a success nor failure
-          console.error(e);
-        }
-      }
+      this.measureDrawLoop(DRAW_LOOP_MEASURE, () => {
```
Comment (joshlory) [inline apps/src/gamelab/GameLab.js:1501]: Are you confident this won't affect Game Lab perf?

File: apps/src/gamelab/GameLab.js [hunk 3]
```diff
@@ -1482,6 +1512,31 @@ GameLab.prototype.captureInitialImage = function () {
   captureThumbnailFromCanvas(document.getElementById('defaultCanvas0'));
 };
 
+/**
+ * Log some performance numbers to firehose
+ */
+GameLab.prototype.reportMetrics = function () {
```
Comment (joshlory) [inline apps/src/gamelab/GameLab.js:1537]: Should we sample the data at all, or fine to log times for every run by every user?

File: apps/src/gamelab/GameLab.js (hunk 1 above)
Comment (joshlory) [reply apps/src/gamelab/GameLab.js]: Ah I see, we divide by number of frames when reporting 👍.

File: apps/src/gamelab/GameLab.js (hunk 2 above)
Comment (balderdash) [reply apps/src/gamelab/GameLab.js:1501]: Did some basic testing, and found that as long as I clear the previous marks every loop, its performance impact is negligible. A million mark/measure calls take ~3 seconds.
image

File: apps/src/gamelab/GameLab.js (hunk 3 above)
Comment (balderdash) [reply apps/src/gamelab/GameLab.js:1537]: Yeah sounds good. Sampling 5% of page loads.

**Comment by joshlory** [comment]
[Add basic Fire Hose instrumentation to Sprite Lab](https://trello.com/c/nriNCgFI/202-add-basic-fire-hose-instrumentation-to-sprite-lab)

**Pull request merged**